cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetsUtil.java

```diff
@@ -134,4 +134,13 @@ public static DatasetSpecification fixOriginalProperties(@Nullable DatasetSpecif
     }
     return spec.setOriginalProperties(props);
   }
+
+  //TODO: CDAP-4627 - Figure out a better way to identify system datasets in user namespaces
+  public static boolean isUserDataset(Id.DatasetInstance datasetInstanceId) {
```

Comment:
There's a JIRA to improve that right?
This PR is just to at least use this unclean way.